precompile/stateful_contract.go

```diff
@@ -0,0 +1,31 @@
+package precompile
+
+import (
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+)
+
+type statefulPrecompiledContract struct {
+	abi abi.ABI
+}
+
+func NewStatefulPrecompiledContract(abiStr string) StatefulPrecompiledContract {
+	abi, err := abi.JSON(strings.NewReader(abiStr))
+	if err != nil {
+		panic(err)
+	}
+	return &statefulPrecompiledContract{
+		abi: abi,
+	}
+}
+
+func (spc *statefulPrecompiledContract) GetABI() abi.ABI {
+	return spc.abi
+}
+
+func (spc *statefulPrecompiledContract) RequiredGas(input []byte) uint64 {
```

Comment:
the contract has multiple methods, shouldn't gas be defined for each method or specifically each state access?

Reply:
you technically can with this if you decode input, but that's some heavy lifting every time you want to have different gas for each method. I have yet to come up with a good way to allow each method to define it's own gas that doesn't use some reflect black magic. Open to ideas